internal/terraform/node_resource_abstract_instance.go

```diff
@@ -2935,6 +2938,62 @@ func (n *NodeAbstractResourceInstance) validateIdentity(newIdentity cty.Value) (
 	return diags
 }
 
+func (n *NodeAbstractResourceInstance) validateIdentityMatchesSchema(newIdentity cty.Value, identitySchema *configschema.Object) (diags tfdiags.Diagnostics) {
```

Comment:
It's unclear to me why this is in a separate validate function. I feel like it makes sense to do this validation in the validateIdentity function, which currently only checks for marks, by adding the schema to the signature. Is there a reason you made this a separate call I'm not seeing? (it's not necessarily wrong in its own function, these are just closely related such that if they always happen together - let's just put it in the same validate)

Reply:
lgtm!! much clearer to review <3